src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,375 @@
+            case XPRESS_AGENCY_ID:
+                return RideType.XPRESS;
+            case MARTA_AGENCY_ID:
+                return RideType.MARTA;
+            case GCT_AGENCY_ID:
+                if(shortName.equalsIgnoreCase("102") ||
+                    shortName.equalsIgnoreCase("103A") ||
+                    shortName.equalsIgnoreCase("110")) {
+                    return RideType.GCT_EXPRESS_Z1;
+                } else if(shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("103")) {
+                    return RideType.GCT_EXPRESS_Z2;
+                }
+                return RideType.GCT_LOCAL;
+            case STREETCAR_AGENCY_ID:
+                return RideType.STREETCAR;
+        }
+        // CobbLinc
+        if (ride.feedId.equals("2")) return RideType.COBB_LOCAL;
+
+        // Xpress
+        else if (ride.feedId.equals("6")) {
+            // No need to differentiate between different XPRESS zones because the xfer policies are the same
+            return RideType.XPRESS;
+        }
+
+        // Default to MARTA
+        return RideType.MARTA;
+        // TODO: Finish
+    }
+
+    private static TransferMeta classifyTransfer(RideType toRideType, RideType fromRideType, Fare.FareType fareType) {
+        switch (toRideType) {
+            case STREETCAR:
+                return new TransferMeta(TransferType.NO_TRANSFER);
+            case COBB_LOCAL:
+                if(!isElectronicPayment(fareType)) {
+                    if (fromRideType == RideType.COBB_LOCAL) {
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    }
+                    return new TransferMeta(TransferType.END_TRANSFER);
+                }
+                switch(fromRideType) {
+                    case COBB_LOCAL:
+                    case COBB_EXPRESS:
+                    case MARTA:
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    default:
+                        return new TransferMeta(TransferType.END_TRANSFER);
+                }
+            case COBB_EXPRESS:
+                if(!isElectronicPayment(fareType)) return new TransferMeta(TransferType.END_TRANSFER);
+                switch(fromRideType) {
+                    case COBB_EXPRESS:
+                    case MARTA:
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    case COBB_LOCAL:
+                        return new TransferMeta(TransferType.TRANSFER_PAY_DIFFERENCE, 180, 4);
+                    default:
+                        return new TransferMeta(TransferType.NO_TRANSFER);
+                }
+            case MARTA:
+               if(!isElectronicPayment(fareType)) return new TransferMeta(TransferType.END_TRANSFER);
+               switch(fromRideType) {
+                   case MARTA:
+                   case XPRESS:
+                   case COBB_LOCAL:
```

Comment:
This is a problem with the streetcar too, but I thought it would never happen in the real world so it was ok. But I wasn't aware that Cobb has a free service as well, so I will need to find a new way to handle this case.

Reply:
I think this is totally fixed. I enabled the relevant streetcar test, and I added a new test for the Cobb circulator bus.